src/robotFactory.js

```diff
@@ -1,17 +1,71 @@
 'use strict';
 
-function BaseRobot() {
+function BaseRobot(name, weight, coords, chipVersion) {
   // implement
+  this.name = name;
+  this.weight = weight;
+  this.chipVersion = chipVersion;
+  this.coords = coords;
 }
 
-function FlyingRobot() {
+BaseRobot.prototype.goForward = function(step = 1) {
+  this.coords.y += step;
+};
+
+BaseRobot.prototype.goBack = function(step = 1) {
+  this.coords.y -= step;
+};
+
+BaseRobot.prototype.goRight = function(step = 1) {
+  this.coords.x += step;
+};
+
+BaseRobot.prototype.goLeft = function(step = 1) {
+  this.coords.x -= step;
+};
+
+BaseRobot.prototype.getInfo = function() {
+  return `Robot: ${this.name},`
+  + ` Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+};
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  BaseRobot.call(this, name, weight, coords, chipVersion);
+  this.coords.z = 0;
   // implement
 }
 
-function DeliveryDrone() {
+FlyingRobot.prototype.goUp = function(step = 1) {
+  this.coords.z += step;
+};
+
+FlyingRobot.prototype.goDown = function(step = 1) {
+  this.coords.z -= step;
+};
+
+Object.setPrototypeOf(FlyingRobot.prototype, BaseRobot.prototype);
+
+function DeliveryDrone(name, weight, coords, chipVersion,
+  maxLoadWeight, currentLoad) {
```

Comment:
переноси аргументы на новую строку
```suggestion
function DeliveryDrone(
  name, weight, coords, chipVersion,
  maxLoadWeight, currentLoad,
) {
```